client/src/components/ShrinkImage/index.tsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import { Transition } from 'react-transition-group';
+import { Theme, createStyles, WithStyles, withStyles } from '@material-ui/core';
+import classNames from 'classnames';
+
+type ShiftedClassName =
+  | 'entering-shift'
+  | 'entered-shift'
+  | 'exiting-shift'
+  | 'exited-shift'
+  | 'unmounted-shift';
+
+const styles = (theme: Theme) =>
+  createStyles({
+    shrinkimage: {
+      display: 'block',
+      height: '100%',
+      maxWidth: '80vw',
+      maxHeight: '80vh',
+      margin: `${theme.spacing(4)}px auto`,
+      transition: 'transform 300ms ease-in-out',
+    },
+    entering: {
+      transform: 'scale(1.25)',
+    },
+    'entering-shift': {
+      transform: 'translateY(100px) scale(1.25)',
+    },
+    entered: {
+      transform: 'scale(1)',
+    },
+    'entered-shift': {
+      transform: 'translateY(0) scale(1)',
+    },
+    exiting: {},
+    'exiting-shift': {},
+    exited: {},
+    'exited-shift': {},
+    unmounted: {},
+    'unmounted-shift': {},
```

Comment:
These are just stubs so that TS doesn't complain when I index `classes` by `state`.

Reply:
makes sense